anki-editor.el

```diff
@@ -1092,6 +1304,21 @@ Return a list of cons of (FIELD-NAME . FIELD-CONTENT)."
            (length anki-editor--note-markers) (buffer-name) (point))
   (push (point-marker) anki-editor--note-markers))
 
+(cl-defun anki-editor--draw-progress-bar (title count total &optional (errors 0) (width 30))
+  "Draw a progress bar."
+  (let ((progress (/ (float count) total)))
+    (message "%s [%s%s] %d/%d (%.2f%%)%s"
+             title
+             (make-string (truncate (* width progress)) ?#)
+             (make-string (- width (truncate (* width progress))) ?.)
+             count
+             total
+             (* 100 progress)
+             (if (zerop errors)
+                 ""
+               (propertize (format " %d errors" errors)
+                           'face `(:foreground "red"))))))
+
 (defun anki-editor-push-notes (&optional scope match &rest skip)
   "Build notes from headings that MATCH within SCOPE and push them to Anki.
 
```

Comment:
If we only change this here, then this will only work on new notes, and in particular only when using `anki-editor-push-new-notes`. However, there are many other types of pushes to notes (at point, subtree, whole buffer, etc.) which should also be supported. This in particular should include "updating" existing notes, in the sense that they should get a hash in their properties drawer

Reply:
I wrote a comment earlier but apparently I blew it away somehow. I'm not super familiar with github's interface so I apologize for any inconvenience. 

I believe -push-note-at-point is the only independent "push", the others all go through -push-notes, so that should handle all those. I added a commit to set the hash on successful -push-note-at-point to bring it in line with the others. Let me know if that resolves your concern or if I misunderstood.

Thanks for all your work looking through this btw. This package has been so much nicer of an interface to anki card-creation than anki itself.

Reply:
Oh indeed, I must've mislooked on some of the other functions; thanks!